src/block/Beacon.php

```diff
@@ -44,61 +51,52 @@ final class Beacon extends Transparent{
 		//TODO netherite block
 	];
 
-	private const ALLOWED_PRIMARY_EFFECTS = [
-		EffectIds::HASTE => true,
-		EffectIds::JUMP_BOOST => true,
-		EffectIds::RESISTANCE => true,
-		EffectIds::SPEED => true,
-		EffectIds::STRENGTH => true
-	];
-
-	private const ALLOWED_SECONDARY_EFFECTS = [
-		EffectIds::REGENERATION => true
-	];
-
-	private int $primaryEffect;
-	private int $secondaryEffect;
+	private Effect $primaryEffect;
+	private ?Effect $secondaryEffect;
 
 	public function readStateFromWorld() : void{
 		parent::readStateFromWorld();
 		$tile = $this->position->getWorld()->getTile($this->position);
 		if($tile instanceof TileBeacon){
-			$this->primaryEffect = $tile->getPrimaryEffect();
-			$this->secondaryEffect = $tile->getSecondaryEffect();
+			$effectIdMap = EffectIdMap::getInstance();
+			$primaryEffectId = $tile->getPrimaryEffect();
+			$primaryEffect = $effectIdMap->fromId($primaryEffectId);
+			if($primaryEffect === null){
+				throw new InvalidBlockStateException("Invalid primary effect $primaryEffectId");
```

Comment:
The beacon may not yet have been activated.

Reply:
Indeed, stupid me